### DISCUSSION TITLE
New diesel guides

### DISCUSSION THREAD

**Discussion by weiznich** [diesel-rs/diesel#3671] [Polls]
I plan to spend some time writing new guides for diesel on currently uncovered topics. I want to use this poll to get an impression guides on which topics are missing according to our users. See [here](https://diesel.rs/guides/) for a list of existing guides. If you have a suggestion for a topic not listed in the poll below, please leave a comment.

**Comment by Th3Whit3Wolf** [comment]
I would appreciate examples showing how to implement keyset and cursor based pagination. I've seen the offset based pagination in the extending diesel section of the guide but I'm not sure how to make something similar with other pagination methods.

**Comment by thkoch2001** [comment]
I like to learn things bottom-up, e.g. I learned Git starting with the 4 object types (BLOB, COMMIT, TREE, TAG), refs and how they are represented in the .git folder.

Likewise I spent yesterday trying to identify a similar basic core of diesel, but I got lost.

Actually I think it would have helped a lot to have more rustdoc comments for basic diesel code.

**Reply by weiznich** [reply]
We are always happy to receive contributions to improve the documentation. Unfortunately there is just to much todo for us maintainers to do all the desired improvements on our own. We are aware that the documentation can be improved, but there is currently just no capacity to do that in a structured way.
